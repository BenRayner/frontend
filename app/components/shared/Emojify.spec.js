/* global describe, it, jest, expect */
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

describe('Emojify', () => {
  const mockParse = jest.fn((text) => text.replace(/:/g, ''));

  jest.mock('../../lib/parseEmoji', () => mockParse);

  const { default: Emojify } = require('./Emojify');

  it('renders `text` with emoji as expected', () => {
    const FIXTURE_TEXT = "This text has :sparkles:emoji:buildkite:, and should be reflected in the title attribute. The text content should contain different text, as processed by lib/parseEmoji!";

    const component = ReactTestRenderer.create(
      <Emojify
        text={FIXTURE_TEXT}
        className="test-class-name"
        updateFrequency={0}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockParse).toHaveBeenCalledWith(FIXTURE_TEXT);
  });

  it('renders plain `text` as expected', () => {
    const FIXTURE_TEXT = "This text should be reflected in the title attribute and text content, as well as in the call to lib/parseEmoji!";

    const component = ReactTestRenderer.create(
      <Emojify
        text={FIXTURE_TEXT}
        className="test-class-name"
        updateFrequency={0}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockParse).toHaveBeenCalledWith(FIXTURE_TEXT);
  });

  it('accepts a `title` override', () => {
    const component = ReactTestRenderer.create(
      <Emojify
        text="This is the text I want to display, it has bee..."
        title="This is the text I want to display, it has been truncated, so the title is used to show the rest on hover!"
        className="another-test-class-name"
        updateFrequency={0}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockParse).toHaveBeenCalledWith("This is the text I want to display, it has bee...");
  });
});
