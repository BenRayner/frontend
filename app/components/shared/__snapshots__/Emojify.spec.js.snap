// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emojify accepts a \`title\` override 1`] = `
<span
  className="another-test-class-name"
  style={undefined}
  title="This is the text I want to display, it has been truncated, so the title is used to show the rest on hover!"
>
  This is the text I want to display, it has bee...
</span>
`;

exports[`Emojify renders \`text\` with emoji as expected 1`] = `
<span
  className="test-class-name"
  dangerouslySetInnerHTML={
    Object {
      "__html": "This text has sparklesemojibuildkite, and should be reflected in the title attribute. The text content should contain different text, as processed by lib/parseEmoji!",
    }
  }
  style={undefined}
  title="This text has :sparkles:emoji:buildkite:, and should be reflected in the title attribute. The text content should contain different text, as processed by lib/parseEmoji!"
/>
`;

exports[`Emojify renders plain \`text\` as expected 1`] = `
<span
  className="test-class-name"
  style={undefined}
>
  This text should be reflected in the title attribute and text content, as well as in the call to lib/parseEmoji!
</span>
`;
